### DISCUSSION TITLE
Access Traces via Lua and/or a TracesToLogs filter

### DISCUSSION THREAD

**Discussion by mp3monster** [fluent/fluent-bit#8711] [Ideas]
It would bev really good if we could access Trace data within a Fluent Bit pipeline. While traces should be immutable, they may well contain meaningful data from which additional metrics can be drawn. Such as how long it takes to complete a span (representing a transactional scope). Having metrics like this will make it possible to start seeing correlations in performance.
There may also be data in the trace relevant to log related activities such as perhaps request originators etc.

The two easiest routes to this would be to create a log event from the trace data and to have access the trace construct through a Lua method - rather than change the existing callback we could have Lua callbacks for each signal type, so the record object is suitably structured?

**Comment by patrick-stephens** [comment]
I think there is some work in this area for processors rather than filters to work with metrics and traces. Currently filters only work for logs.
